src/RawRabbit.Enrichers.MessagePack/MessagePackSerializerWorker.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Linq;
+using System.Reflection;
+using MessagePack;
+using RawRabbit.Serialization;
+
+namespace RawRabbit.Enrichers.MessagePack
+{
+	internal class MessagePackSerializerWorker : ISerializer
+	{
+		public string ContentType => "application/x-messagepack";
+		private readonly MethodInfo _deserializeType;
+		private readonly MethodInfo _serializeType;
+
+		public MessagePackSerializerWorker(bool useLz4)
```

Comment:
Does `LZ4MessagePackSerializer` and `MessagePackSerializer` share the same base class? If so, it would be neater if the constructor here would accept that here. That way we can get rid of the reflection in the contructor

Reply:
They don't - they're both static.

Reply:
I'm also unsure how to rid the reflection. The issue is that we need to call a generic method always, and RawRabbit works with both generic and non-generic method calls. So to handle the non-generics (one serialize, one deserialize), the generation of methods is needed.

I'm certain the reflection prevents this from being super fast, but it should definitely be useable.